EL-4088 - Use currently active theme in plunkers

DeanHaire opened the pull request:
#### Checklist
<!-- Use an 'x' to check those which apply. -->
* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue
<!-- Either a Jira URL or a description of the issue that this PR addresses. -->
https://portal.digitalsafe.net/browse/EL-4088

#### Description of Proposed Changes
<!-- Describe what was changed and how it addresses the original issue. -->
- Load the appropriate stylesheets for the currently active theme within plunker examples
- Include Roboto when using the 2020 theme

#### Documentation CI URL
<!-- Initiate a build at https://jenkins.swinfra.net/job/SEPG/view/Templates/job/New%20SEPG%20Build/build -->
<!-- Append the branch name to the following URL: -->
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4088-roboto-plunker-themes

#### Downstream Build(s)
<!-- Push a branch with the same name in each downstream repository and create a build in Jenkins. -->
<!-- Add the Jenkins build link(s) below: -->
* [caf/ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4088-roboto-plunker-themes~CI/)

buildmachine-sou-jenkins2 commented:
CI Build Link:
https://jenkins.swinfra.net/job/SEPG/job/UXAspects/job/UXAspects~UXAspects~EL-4088-roboto-plunker-themes~CI

buildmachine-sou-jenkins2 commented:
The Documentation QA site for this branch has been built:
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4088-roboto-plunker-themes

donna-marijne reviewed (changes requested):
This does get the job done, albeit with duplicated code. Perhaps extending the existing strategy pattern would be cleaner?
```ts
    private createPlaygroundStrategy(playground: IPlayground, theme: SiteTheme): PlaygroundStrategy {
        switch (playground.framework) {
            case 'css':
                return new CssPlaygroundStrategy(this._documentationType, this.createThemeStrategy(theme));
            case 'angular':
            default:
                return new AngularPlaygroundStrategy(this._documentationType, this.createThemeStrategy(theme));
        }
    }
```

```ts
return [...stylesheets, ...this._themeStrategy.getGlobalExternalStyles(assetsUrl)];
```

Also consider pulling the SiteThemeService up from the MF repo so that we don't duplicate that code either.

DeanHaire commented:
I've removed the duplicate code and simplified the implementation as much as I can. I've also lifted up the site theme service to make better use of that.

donna-marijne commented on docs/app/services/playground/strategies/playground-strategy.ts:
```diff
@@ -31,7 +34,28 @@ export abstract class PlaygroundStrategy {
     /** Get all the external styles that should be loaded in the `head` of the HTML */
     getGlobalExternalStyles(assetsUrl: string): string[] {
-        return [];
+        const stylesheets = [
```
I don't think this should be added to the base abstract class. I think you can extend the strategy pattern as described previously to keep it clean. Let me know what was unclear and I can try to explain further.

donna-marijne commented on docs/app/app.component.ts:
```diff
@@ -89,8 +90,10 @@ export class AppComponent implements OnInit {
+        ngZone: NgZone,
+        siteThemeService: SiteThemeService) {
         (<any>window).ngZone = ngZone;
+        siteThemeService.load();
```
We don't actually need to load the service in the open source site as far as I'm aware. The service just needs to be here because all the playground code is here too. The MF site has it's own independent AppComponent.

donna-marijne commented on docs/app/services/site-theme/site-theme.service.ts:
```diff
@@ -0,0 +1,56 @@
+
+    load(): void {
+
+        if (this._documentationType !== DocumentationType.Keppel) {
```
Can be removed if you avoid calling this function in the open source site.

donna-marijne commented on docs/app/services/playground/strategies/themes/microfocus-2020-strategy.ts:
```diff
@@ -0,0 +1,13 @@
+    getStylesheets(assetsUrl: string): string[] {
+        return [
+            SystemJSHelper.getPackageUrl({ name: 'bootstrap', path: 'bootstrap@3.3.7/dist/css/bootstrap.min.css' }),
+            `${assetsUrl}/styles/ux-aspects.css`,
```
Since you've added a KeppelThemeStrategy, you could extend that for the three Micro Focus themes and eliminate this bit of duplication as well.

DeanHaire replied:
I don't think I can, the `ux-aspects.css` stylesheet is in different locations. i.e. `/css/ux-aspects.css` for keppel and `/styles/ux-aspects.css` for all micro focus themes

donna-marijne replied:
Good point, don't worry about it then.

donna-marijne reviewed (changes requested):
It looks good and works well, there's just a bit of cleanup.

donna-marijne reviewed (approved):
Looks good, thank you.
